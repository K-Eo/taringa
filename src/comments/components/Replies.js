import React from 'react'
import { compose } from 'recompose'

import withLoadMore from '../../HOC/withLoadMore'
import Reply from './ReplyContainer'

export const Replies = ({ items }) => {
  return items.map(i => <Reply key={i} id={i} />)
}

Replies.defaultProps = {
  items: [],
}

const getShowLoader = props => props.status === 'fetching'
const getHasMore = props => props.items.length < props.totalCount
const getHasFailure = props => props.status === 'failure'

export default compose(withLoadMore(getShowLoader, getHasMore, getHasFailure))(
  Replies
)
