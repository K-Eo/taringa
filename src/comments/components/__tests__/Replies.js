import React from 'react'

import Replies from '../Replies'

describe('Replies', () => {
  it('exists', () => {
    expect(Replies).toBeDefined()
  })

  it('renders default', () => {
    const wrapper = shallow(<Replies />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders toggler when has more than item', () => {
    const props = {
      items: ['foo'],
      totalCount: 1,
    }

    const wrapper = shallow(<Replies {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders items on toggle visibility', () => {
    const props = {
      items: ['foo', 'bar'],
      totalCount: 2,
    }

    const wrapper = shallow(<Replies {...props} />)
    expect(wrapper).toMatchSnapshot()

    wrapper.find('button.Commentable-replies-toggler').simulate('click')

    expect(wrapper).toMatchSnapshot()

    wrapper.find('button.Commentable-replies-toggler').simulate('click')

    expect(wrapper).toMatchSnapshot()
  })
})
