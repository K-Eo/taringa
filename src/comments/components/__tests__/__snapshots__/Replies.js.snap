// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Replies renders default 1`] = `<React.Fragment />`;

exports[`Replies renders items on toggle visibility 1`] = `
<React.Fragment>
  <button
    className="btn btn-link text-dark Commentable-replies-toggler mb-2"
    onClick={[Function]}
  >
    <span>
      Ver 2 respuestas 
      <i
        className="fa fa-chevron-down"
      />
    </span>
  </button>
</React.Fragment>
`;

exports[`Replies renders items on toggle visibility 2`] = `
<React.Fragment>
  <button
    className="btn btn-link text-dark Commentable-replies-toggler mb-2"
    onClick={[Function]}
  >
    <span>
      Ocultar respuestas 
      <i
        className="fa fa-chevron-up"
      />
    </span>
  </button>
  <Connect(Reply)
    id="foo"
    key="foo"
  />
  <Connect(Reply)
    id="bar"
    key="bar"
  />
</React.Fragment>
`;

exports[`Replies renders items on toggle visibility 3`] = `
<React.Fragment>
  <button
    className="btn btn-link text-dark Commentable-replies-toggler mb-2"
    onClick={[Function]}
  >
    <span>
      Ver 2 respuestas 
      <i
        className="fa fa-chevron-down"
      />
    </span>
  </button>
</React.Fragment>
`;

exports[`Replies renders toggler when has more than item 1`] = `
<React.Fragment>
  <button
    className="btn btn-link text-dark Commentable-replies-toggler mb-2"
    onClick={[Function]}
  >
    <span>
      Ver respuesta 
      <i
        className="fa fa-chevron-down"
      />
    </span>
  </button>
</React.Fragment>
`;
